backend/fms_core/services/sample_next_step.py

```diff
@@ -48,4 +48,125 @@ def queue_sample_to_study_workflow(sample_obj: Sample, study_obj: Study, order:
+        except Exception as err:
+            errors.append(err)
+    return num_deleted, errors, warnings
+
+def dequeue_sample_from_all_steps_study_workflow(sample_obj: Sample, study_obj: Study) -> Tuple[Union[int, None], List[str], List[str]]:
+    """
+    Deletes a SampleNextStep instance to indicate the removal of a sample from a specified workflow at any step. 
+    Meaning that if the sample is queued in 2 different steps in the provided study, both instances will be deleted. 
+    Args:
+        `sample_obj`: Sample instance to be dequeued from the study workflow.
+        `study_obj`: Study instance currently associated to the sample.
+                 
+    Returns:
+        Tuple containing the number of SampleNextStep instances deleted, the error messages and the warning messages. 
+
+    """
+    dequeued = None
+    errors = []
+    warnings = []
+
+    if not isinstance(sample_obj, Sample):
+        sample_obj = None
+        errors.append(f"A valid sample instance must be provided.")
+
+    if not isinstance(study_obj, Study):
+        study_obj = None
+        errors.append(f"A valid study instance must be provided.")
+    
+    # Dequeuing from study workflow by deleting the SampleNextStep instance
+    if sample_obj and study_obj and not errors:
+        try:
+            num_deleted, _ = SampleNextStep.objects.filter(sample=sample_obj, study=study_obj).delete()
+        except Exception as err:
+            errors.append(err)
+    return num_deleted, errors, warnings
+
+def is_sample_queued_in_study(sample_obj: Sample, study_obj: Study, order: int=None) -> Tuple[Union[bool, None], List[str], List[str]]:
+    """
+    
+
+    Args:
+        `sample_obj`: Sample instance to look for.
+        `study_obj`: Study instance where to look the sample in.
+        `order`: Positive integer indicating the current position in the workflow. Optional: If not specified then it will look at any step. 
+                 
+    Returns:
+        Tuple containing a boolean stating whether the sample is queued in the study, the error messages and the warning messages. 
+
+    """
+    sample_is_queued = None
+    errors = []
+    warnings = []
+
+    if not isinstance(sample_obj, Sample):
+        sample_obj = None
+        errors.append(f"A valid sample instance must be provided.")
+
+    if not isinstance(study_obj, Study):
+        study_obj = None
+        errors.append(f"A valid study instance must be provided.")
+    
+    step_order=None
+    if order is not None:
+        try:
+            step_order = StepOrder.objects.get(order=order, workflow=study_obj.workflow)
+        except StepOrder.DoesNotExist:
+            errors.append(f"No step found for the given order.")
+    
+    # Specify step_order in filter if it is provided
+    filters = dict(
+        sample=sample_obj,
+        study=study_obj,
+        **(dict(step_order=step_order) if step_order is not None else dict())
```

Comment:
what if i want to know if a sample completed a study workflow ... the expected step order would be none. The function would assume i look for any order... might be a good idea to either split the function or to add another boolean param to change the behaviour in a predictable way.